pkg/registry/common/setid/server.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2020-2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package setid
+
+import (
+	"context"
+	"strings"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/google/uuid"
+
+	"github.com/networkservicemesh/api/pkg/api/registry"
+
+	"github.com/networkservicemesh/sdk/pkg/registry/core/next"
+)
+
+type setIDServer struct {
+	names namesSet
+}
+
+// NewNetworkServiceEndpointRegistryServer creates new instance of NetworkServiceRegistryServer which set the unique
+// name for the endpoint on registration
+func NewNetworkServiceEndpointRegistryServer() registry.NetworkServiceEndpointRegistryServer {
+	return new(setIDServer)
+}
+
+func (s *setIDServer) Register(ctx context.Context, nse *registry.NetworkServiceEndpoint) (*registry.NetworkServiceEndpoint, error) {
+	name, suffix := interDomainName(nse.Name)
```

Comment:
Why setid knows about interdomain?

Reply:
Because if we want to register NSE in some other domain, we shouldn't override domain suffix:
https://github.com/networkservicemesh/sdk/blob/536f38f8986817e62259be5f258203006ef96a7e/pkg/registry/core/interdomain/interdomain_nse_test.go#L239-L246

Reply:
Yes, but this chain element shouldn't know about the interdomain.

Reply:
Fixed with id prefix.